android/java/org/chromium/chrome/browser/crypto_wallet/activities/BraveWalletActivity.java

```diff
@@ -205,73 +186,6 @@ public ModalDialogManager getModalDialogManager() {
-        } else if (walletAction == WalletAction.UNLOCK) {
-            UnlockWalletFragment unlockWalletFragment = new UnlockWalletFragment();
-            navigationFragments.add(unlockWalletFragment);
-        } else if (walletAction == WalletAction.ONBOARDING_RESTORE) {
-            mShowBiometricPrompt = false;
-            OnboardingRestoreWalletFragment onboardingRestoreWalletFragment =
-                    OnboardingRestoreWalletFragment.newInstance();
-            navigationFragments.add(onboardingRestoreWalletFragment);
-        }
-
-        if (mCryptoWalletOnboardingPagerAdapter != null) {
-            mCryptoWalletOnboardingPagerAdapter.setNavigationItems(navigationFragments);
-        }
-        addRemoveSecureFlag(true);
-    }
-
-    @SuppressLint("NotifyDataSetChanged")
-    private void replaceNavigationFragments(@NonNull final WalletAction walletAction) {
-        if (mCryptoWalletOnboardingViewPager == null) return;
-        if (mCryptoWalletOnboardingPagerAdapter == null) return;
-
-        final List<BaseWalletNextPageFragment> navigationFragments = new ArrayList<>();
-        // Terms of use screen is shown only during onboarding actions.
-        if (walletAction != WalletAction.RESTORE) {
-            final OnboardingTermsOfUseFragment onboardingTermsOfUseFragment =
-                    OnboardingTermsOfUseFragment.newInstance();
-            navigationFragments.add(onboardingTermsOfUseFragment);
-            final OnboardingNetworkSelectionFragment onboardingNetworkSelectionFragment =
-                    OnboardingNetworkSelectionFragment.newInstance();
-            navigationFragments.add(onboardingNetworkSelectionFragment);
-        }
-
-        if (walletAction == WalletAction.ONBOARDING_RESTORE
-                || walletAction == WalletAction.RESTORE) {
-            mShowBiometricPrompt = false;
-
-            final OnboardingRestoreWalletFragment onboardingRestoreWalletFragment =
-                    OnboardingRestoreWalletFragment.newInstance();
-            navigationFragments.add(onboardingRestoreWalletFragment);
-            addWalletCreationPage(navigationFragments);
-
-        } else if (walletAction == WalletAction.PASSWORD_CREATION) {
-            mShowBiometricPrompt = true;
-
-            final OnboardingSecurePasswordFragment onboardingSecurePasswordFragment =
-                    new OnboardingSecurePasswordFragment();
-            navigationFragments.add(onboardingSecurePasswordFragment);
-            addWalletCreationPage(navigationFragments);
-            addBackupWalletSequence(navigationFragments, true);
-        }
-        mCryptoWalletOnboardingPagerAdapter.replaceWithNavigationItems(
-                navigationFragments, mCryptoWalletOnboardingViewPager.getCurrentItem() + 1);
-        mCryptoWalletOnboardingPagerAdapter.notifyDataSetChanged();
-
-        mCryptoWalletOnboardingViewPager.setCurrentItem(
-                mCryptoWalletOnboardingViewPager.getCurrentItem() + 1);
-    }
-
```

Comment:
This entire logic has been replaced by a `position -> fragment` algorithm in `WalletOnboardingPagerAdapter#createFragment`. Doing so, the fragments are recycled when needed and their state is handled directly by the recycler, making them more memory efficient.